src/Data/String/CodePoints.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+/* global Symbol */
+
+var hasArrayFrom = typeof Array.from === "function";
+var hasStringIterator =
+  typeof Symbol !== "undefined" &&
+  Symbol != null &&
+  typeof Symbol.iterator !== "undefined" &&
+  typeof String.prototype[Symbol.iterator] === "function";
+var hasFromCodePoint = typeof String.prototype.fromCodePoint === "function";
+var hasCodePointAt = typeof String.prototype.codePointAt === "function";
+
+exports._unsafeCodePointAt0 = function (fallback) {
+  return hasCodePointAt
+    ? function (str) { return str.codePointAt(0); }
+    : fallback;
+};
+
+exports._codePointAt = function (fallback) {
+  return function (Just) {
+    return function (Nothing) {
+      return function (unsafeCodePointAt0) {
+        return function (index) {
+          return function (str) {
+            var length = str.length;
+            if (index < 0 || index >= length) return Nothing;
+            if (hasStringIterator) {
+              var iter = str[Symbol.iterator]();
+              for (var i = index;; --i) {
+                var o = iter.next();
+                if (o.done) return Nothing;
+                if (i === 0) return Just(unsafeCodePointAt0(o.value));
+              }
+            } else if (hasArrayFrom) {
```

Comment:
Since, according to MDN, String iteration is not as well supported as `Array.from`, having `_codePointAt` first try String iteration, then `Array.from`, and then the PureScript fallback means that `codePointAt` might now be linear in either the index or the length of the entire string, depending on what's supported on the platform we're running on. I'm not necessarily opposed to using a heuristic to decide whether to use `Array.from`, but I don't think we should have the function's complexity depending on different things based on the platform. Can we just remove `Array.from` until the heuristic question has been better addressed please?